src/main/java/net/dv8tion/jda/internal/requests/ratelimit/BotRateLimiter.java

```diff
@@ -266,11 +266,12 @@ else if (cloudflare)
                     // Handle hard rate limit, pretty much just log that it happened
                     else
                     {
-                        boolean firstHit = hitRatelimit.add(baseRoute);
+                        boolean firstHit = retryAfter >= 6000 || hitRatelimit.add(baseRoute);
```

Comment:
Should this be `<`? We want to log it as debug only if it's *not* long or we haven't hit it before if I understand correctly.

Reply:
Good catch, my mistake.
```suggestion
                        boolean firstHit = retryAfter < 6000 || hitRatelimit.add(baseRoute);
```